commit cb6ca8e1ed922082bacc6e5e5ee040491a443ea2
Author: Johan Hedberg <johan.hedberg@intel.com>
Date:   Mon Jan 6 18:27:02 2014 +0200

    Bluetooth: Default to no security with L2CAP RAW sockets
    
    L2CAP RAW sockets can be used for things which do not involve
    establishing actual connection oriented L2CAP channels. One example of
    such usage is the l2ping tool. The default security level for L2CAP
    sockets is LOW, which implies that for SSP based connection
    authentication is still requested (although with no MITM requirement),
    which is not what we want (or need) for things like l2ping. Therefore,
    default to one lower level, i.e. BT_SECURITY_SDP, for L2CAP RAW sockets
    in order not to trigger unwanted authentication requests.
    
    Signed-off-by: Johan Hedberg <johan.hedberg@intel.com>
    Signed-off-by: Marcel Holtmann <marcel@holtmann.org>

diff --git a/net/bluetooth/l2cap_sock.c b/net/bluetooth/l2cap_sock.c
index e7806e6d282c..20ef748b2906 100644
--- a/net/bluetooth/l2cap_sock.c
+++ b/net/bluetooth/l2cap_sock.c
@@ -147,6 +147,9 @@ static int l2cap_sock_bind(struct socket *sock, struct sockaddr *addr, int alen)
 		    __le16_to_cpu(la.l2_psm) == L2CAP_PSM_RFCOMM)
 			chan->sec_level = BT_SECURITY_SDP;
 		break;
+	case L2CAP_CHAN_RAW:
+		chan->sec_level = BT_SECURITY_SDP;
+		break;
 	}
 
 	bacpy(&chan->src, &la.l2_bdaddr);